forge/csrc/autograd/autograd.cpp

```diff
@@ -484,6 +484,8 @@ Graph *autograd_engine::run()
     create_backward_graph(requires_grad_map);
     create_optimizer_graph();
 
+    insert_recompute_ops();
```

Comment:
Does it make sense to have an `if` statement here to determine if we should insert recompute ops?

Just by briefly passing through the `Graph *autograd_engine::run()` function, on the first go it would seem like we're running a recompute pass each time. In this case, it isn't too big of an issue, but if the amount of passes increases for autograd, it'll be a bit trickier to determine what is run, and what is not.

Reply:
The `if` is inside the function.

Not sure if I understand, but `autograd_engine::run()` is called only once during compile on the main graph...

Reply:
What do you think about doing this:
```cpp
    if (!config.recompute)
    {
        insert_recompute_ops();
    }
```

Looking at other passes it's easier to preview what is called when by first look (e.g. no need to check every function that is called).

Reply:
Not a hard requirement, this is just a personal preference.

Reply:
Done!